src/ota_metadata/legacy/parser.py

```diff
@@ -712,29 +739,19 @@ def _process_text_base_otameta_file(_metafile: MetaFile):
                     self.total_files_num = _count
 
         # ------ start downloading ota_metadata files ------ #
-        start_time = int(time.time())
-
-        def _watchdog_abort_on_no_progress():
-            _last_active_time = max(start_time, self._downloader.last_active_timestamp)
-            if int(time.time()) - _last_active_time > self.download_max_idle_time:
-                logger.error(
-                    f"downloader becomes stuck for {self.download_max_idle_time=} seconds, abort"
-                )
-                raise ValueError
-
         with ThreadPoolExecutorWithRetry(
+            max_workers=2,
             max_concurrent=self.MAX_COCURRENT,
-            watchdog_func=_watchdog_abort_on_no_progress,
         ) as _mapper:
             try:
-                for _ in _mapper.ensure_tasks(
+                for _fut in _mapper.ensure_tasks(
                     _process_text_base_otameta_file,
                     self._ota_metadata.get_img_metafiles(),
                 ):
-                    pass
-            except ValueError as e:
-                logger.error(f"failed to start threadpool: {e!r}")
-                raise
+                    if exc := _fut.exception():
+                        logger.warning(
+                            f"detect failure during downloading OTA image metafiles, retrying: {exc!r}"
+                        )
             except TasksEnsureFailed as e:
                 logger.error(f"faild to finish download all ota metadata files: {e!r}")
                 raise
```

Comment:
Here the idle downloading timeout is removed for downloading the OTA image metadata.

Reply:
We will re-introduce the timeout setting in next PR related to ota_metadata downloading.